src/state/reducers.ts

```diff
@@ -1,10 +1,27 @@
-import {
-  combineReducers, Action, Reducer, ReducersMapObject,
-} from 'redux';
-import query from '../search/state/reducers';
-import results from '../results/state/reducers';
+import { combineReducers, Action, Reducer, ReducersMapObject } from 'redux';
+import query, { SearchAction } from '../search/state/reducers';
+import results, { ResultAction } from '../results/state/reducers';
+import { SearchState } from '../search/state/initialState';
+import { ResultsState } from '../results/state/initialState';
 
-export default combineReducers({
+type RootState =
+  | {
+      query: SearchState;
+      results: ResultsState;
+    }
+  | undefined;
+
+const appReducer = combineReducers({
   query,
   results,
 });
+
+const rootReducer = (state: RootState, action: SearchAction | ResultAction) => {
```

Comment:
Rather than wrapping the combined reducers (`appReducer`), we can pick up `action.type === 'RESET'` in both the search and results reducers instead.

Reply:
Isn't it better to have a single place to do this though, as it affects both parts of the state (and might in the future be needed to reset other parts of the state)?

Reply:
I suppose that depends if we would ever want some specificity in the future - will always want all parts of the app state reset or will there be a time one just some subset reset?

Reply:
We might not want to always reset everything, which is why it's nice to keep our options open.

Reply:
just want to clarify: you're saying leave it as is for now (reset all) and in the future if needed we will replace this PR's code with a state slice specific reset (ie RESET is caught in each of the separate reducers)?

Reply:
No, I mean that in the future we are likely to have more/different RESETS across separate parts of the state so it's better to keep it in a cross slice reducer